### PR TITLE
Single Number II solution modified.

### PR DESCRIPTION
**Pull request by kcterala**
Single Number II solution modified with O(1) space complexity solution.

### CODE REVIEW

File: Bit Manipulation/Single Number II.java [hunk 1]
```diff
@@ -1,21 +1,14 @@
...
-        }
-
-        int ans = -1;
-
-        for (Map.Entry<Integer, Integer> entry : map.entrySet()) {
-            if (entry.getValue() == 1) {
-                ans = entry.getKey();
-                break;
+        int sum = 0;
+        for(int i=0;i<32;i++){
```
Comment (varunu28) [inline Bit Manipulation/Single Number II.java]: Kindly add horizontal space between assignment. The loop should look like:
```
for (int i = 0; i < 32; i++) {
```

https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespacehttps://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace

File: Bit Manipulation/Single Number II.java [hunk 2]
```diff
@@ -1,21 +1,14 @@
...
+            int bitSum =0;
```
Comment (varunu28) [inline Bit Manipulation/Single Number II.java]: Keep spacing consistent in assignment. 
```
int bitSum = 0;
```

File: Bit Manipulation/Single Number II.java [hunk 3]
```diff
@@ -1,21 +1,14 @@
...
+            for(int j=0;j<A.size();j++){
```
Comment (varunu28) [inline Bit Manipulation/Single Number II.java]: Same comment for horizontal spacing.

File: Bit Manipulation/Single Number II.java [hunk 4]
```diff
@@ -1,21 +1,14 @@
...
+                bitSum += ((A.get(j) >> i) & 1);
             }
+            sum += (bitSum%3) * Math.pow(2, i);
```
Comment (varunu28) [inline Bit Manipulation/Single Number II.java]: Please keep spacing consistent.

File: Bit Manipulation/Single Number II.java (hunk 1 above)
Comment (varunu28) [inline Bit Manipulation/Single Number II.java]: Can you please add a comment about why this approach works?

Comment (kcterala) [reply Bit Manipulation/Single Number II.java]: yeah will do now.

Comment (kcterala) [reply Bit Manipulation/Single Number II.java]: Okay

**Comment by kcterala** [comment]
Added comments and example

**Pull request closed without merging**